webui/react/src/pages/TrialDetails/Profiles/SystemMetricChart.tsx

```diff
@@ -37,7 +39,14 @@ const SystemMetricChart: React.FC<Props> = ({ systemMetrics }: Props) => {
             return splits.map(i => dayjs.utc(i).format('HH:mm:ss'));
           },
         },
-        ...systemMetrics.names.map((name) => ({ label: getUnitForMetricName(name) })),
+        {
+          label: getUnitForMetricName(filters.name || ''),
+          size: (self: uPlot, values: string[]) => {
+            if (!values) return 50;
+            const maxChars = Math.max(...values.map(el => el.toString().length));
+            return 25 + Math.max(25, maxChars * 8);
+          },
```

Comment:
Nice, this fixes the overlapping of label and y-axis labels nicely. It's hard coded to 8 pixels for the character width, but we can probably link this to something like label font-size or something to get a more accurate sizing in the future.